include/sys/zil.h

```diff
@@ -395,6 +395,7 @@ typedef struct itx {
 	uint8_t		itx_sync;	/* synchronous transaction */
 	zil_callback_t	itx_callback;   /* Called when the itx is persistent */
 	void		*itx_callback_data; /* User data for the callback */
+	size_t		itx_size;	/* itx size including lr.lrc_reclen */
```

Comment:
the comment here seems a little sketchy to me, since if `lr.lrc_reclen` changes, we won't modify `itx_size` to match this comment. it might be better to state this field is used to ensure we properly free the `itx_t` structure, since that's really the reason the field exists, right? with the comment as is, I might interpret the fact we don't modify this field when changing `lr.lrc_reclen` as a bug, when that's not the case.

Reply:
Good point, right this simply needs to exist because `kmem_free()` requires the size.  Passing a different value for the size will only result in an ASSERT when DEBUG_KMEM_TRACKING is defined.  I'll change it to,

```
	size_t		itx_size;	/* allocated itx struct size */
 ```

Reply:
:+1: great, thanks!